oshmem/mca/memheap/base/memheap_base_alloc.c

```diff
@@ -90,3 +100,30 @@ int mca_memheap_alloc_with_hint(size_t size, long hint, void** ptr)
 
     return MCA_MEMHEAP_CALL(alloc(size, ptr));
 }
+
+map_segment_t *mca_memheap_base_allocate_segment(mca_memheap_map_t *map)
+{
+    map_segment_t *segments;
+    int capacity;
+
+    assert(map->n_segments <= map->capacity);
+
+    if (map->n_segments >= mca_memheap_base_max_segments) {
+        return NULL;
+    }
+
+    if (map->n_segments >= map->capacity) {
```

Comment:
this conflicts with the assert above (line 109)

Reply:
why not checking for `0` here instead of comparing `n_segments` and `capacity`?
currently it looks quite confusing

Reply:
ok, fixed